pkg/endpoint/endpoint.go

```diff
@@ -334,6 +336,10 @@ func (e *Endpoint) LXCMac() mac.MAC {
 	return e.mac
 }
 
+func (e *Endpoint) IsHost() bool {
+	return e.ID == idallocator.HostEndpointID
```

Comment:
Yeah and this function would return true if the endpoint contains the `reserved:host` label

Reply:
I took a slightly different approach here. Instead of checking labels every time we call `IsHost()`, I introduced a new `isHost` boolean in `Endpoint`. When we create the host endpoint we can just set the boolean; when we restore the host endpoint, we can derive it once from labels.

Reply:
FWIW the labels selection should be derived once and cached assuming we use the standard Cilium structures. (This was a high-impact policy optimization a couple of releases back)

Reply:
To match against rules yes, but `Endpoint.HasLabels()` just iterates over labels. Anyway, setting and using a boolean also avoided some locking issues on labels :-)